test/parallel/test-domain-timers-uncaught-exception.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that the timer callbacks are called in the order in which
+// they were created in the event of an unhandled exception in the domain.
+
+const domain = require('domain').create();
+const assert = require('assert');
+
+var first = false;
+
+domain.run(function() {
+  setTimeout(() => { throw new Error('FAIL'); }, 1);
+  setTimeout(() => { first = true; }, 1);
+  setTimeout(() => { assert.strictEqual(first, true); }, 2);
+});
```

Comment:
@jBarz sorry for taking a long time to get back to this.

Could you add something in the end of `domain.run()` that will eat some processor time to ensure that 2ms had always passed and that this test will fail 100% (or 99%) of the time if the patch ever is wrongly changed in the future?

I'm thinking something like the following, which seemed to work just fine (even at `1e5 iterations)` locally:

```js
domain.run(function() {

  // ...

  let i = 1e6;
  while (i--);
});
```